chat-plugins/mafia.js

```diff
@@ -1,919 +1,916 @@
+						break;
+					}
+					role = Object.assign({}, MafiaData.roles[id]);
+					role.baseRole = toId(role.name);
+					role.name = target.slice().map(p => { return toId(p) === 'solo' ? '' : p; }).join(' ');
+					role.id = toId(role.name);
+					if (!role.memo) role.memo = [];
+					role.memo.unshift('During the Day, you may vote for whomever you want lynched.');
+					target.splice(j, 1);
+					j++;
+				}
+			}
+			if (hasProblem) continue;
+			if (!role.name) role = {name: target.join(' '), id: toId(target.join('')), memo: []};
+			// Add modifiers
+			for (let j = 0; j < target.length; j++) {
+				let id = toId(target[j]);
+				if (id.endsWith('shot') && parseInt(target[j].substring(0, 1))) {
+					// Special case for X-Shot modifiers
+					let number = target[j].substring(0, 1);
+					let memo = MafiaData.modifiers.xshot.memo.slice();
+					memo = memo.map(m => { return m.replace(/X/g, number); });
+					role.memo = role.memo.concat(memo);
+					target.splice(j, 1);
+					j--;
+				} else if (MafiaData.modifiers[id]) {
+					if (typeof MafiaData.modifiers[id] === 'string') target[j] = MafiaData.modifiers[id];
+					if (MafiaData.modifiers[id].memo) role.memo = role.memo.concat(MafiaData.modifiers[id].memo);
+					target.splice(j, 1);
+					j--;
+				}
+			}
+			if (hasProblem) continue;
+			// Determine the role's alignment
+			for (let j = 0; j < target.length; j++) {
+				let id = toId(target[j]);
+				if (MafiaData.alignments[id]) {
+					if (typeof MafiaData.alignments[id] === 'string') target[j] = MafiaData.alignments[id];
+					if (role.alignment) {
+						// A role cant have multiple alignments
+						hasProblem = true;
+						problems.push(`The role "${role.name}" has multiple possible alignments (${MafiaData.alignments[id].id} or ${role.alignment})`);
+						break;
+					}
+					role.alignment = MafiaData.alignments[id].id;
+					role.memo = role.memo.concat(MafiaData.alignments[id].memo);
+					if (!role.image && MafiaData.alignments[id].image) role.image = MafiaData.alignments[id].image;
+					target.splice(j, 1);
+					j--;
+				}
+			}
+			if (!role.alignment) role.alignment = 'town'; // Default to town
+			if (alignments.indexOf(role.alignment) === -1) alignments.push(role.alignment);
+			// Handle anything that is unknown
+			if (target.length) {
+				role.memo.push(`To learn more about your role, PM the host (${this.host}).`);
+			}
+			this.originalRoles.push(role);
+		}
+		if (alignments.length < 2) problems.push(`There must be at least 2 different alignments in a game!`);
+		if (problems.length) {
+			this.originalRoles = [];
+			return problems;
+		}
+		this.roles = this.originalRoles.slice();
+		this.originalRoleString = this.originalRoles.slice().map(r => { return `<span style="font-weight:bold;color:${MafiaData.alignments[r.alignment].color || '#FFF'}">${r.name}</span>`; }).join(', ');
+		this.roleString = this.originalRoleString;
+		this.updatePlayers();
+		return [];
```

Comment:
Manually returning an empty array here seems like a strange behavior, could it return ``false`` or ``null`` (preferably) instead?

Reply:
The command that calls this function expects an array to be returned, though I guess I can change it.